voila/static_file_handler.py

```diff
@@ -37,3 +38,21 @@ def get_absolute_path(self, root, path):
                 self.root = root  # make sure all the other methods in the base class know how to find the file
                 break
         return abspath
+
+
+class WhiteListFileHandler(tornado.web.StaticFileHandler):
+    def initialize(self, whitelist=[], blacklist=[], **kwargs):
+        self.whitelist = whitelist
+        self.blacklist = blacklist
+        super(WhiteListFileHandler, self).initialize(**kwargs)
+
+    def get_absolute_path(self, root, path):
+        # StaticFileHandler.get always calls this method first, so we use this as the
+        # place to check the path. Note that now the path seperator is os dependent (\\ on windows)
+        whitelisted = any(re.fullmatch(pattern, path) for pattern in self.whitelist)
+        blacklisted = any(re.fullmatch(pattern, path) for pattern in self.blacklist)
+        if not whitelisted:
+            raise tornado.web.HTTPError(403, 'File not whitelisted')
```

Comment:
I think these probably should be 404s to be indistinguishable from the file not existing. If you decide on 403s, though, perhaps a more generic message is better, like the default 403 message, or just now message.